components/brave_ads/core/internal/serving/ad_serving_util.h

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_CORE_INTERNAL_SERVING_AD_SERVING_UTIL_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_CORE_INTERNAL_SERVING_AD_SERVING_UTIL_H_
+
+#include <vector>
+
+#include "base/check.h"
+#include "base/rand_util.h"
+
+namespace brave_ads {
+
+template <typename T>
+T ChooseCreativeAd(const std::vector<T>& creative_ads) {
+  CHECK(!creative_ads.empty());
+
+  const int rand = base::RandInt(0, static_cast<int>(creative_ads.size()) - 1);
```

Comment:
Probably `base::RandGenerator(creative_ads.size())` is better, as you're passing `rand` to a `std::vector<>::size_type`.

Reply:
Changed